dotCMS/src/main/java/com/dotmarketing/db/DotRunnableThread.java

```diff
@@ -4,63 +4,84 @@
+    try {
+      LocalTransaction.wrap(this::internalRunner);
+    } catch (Exception dde) {
+      throw new DotStateException(dde);
+    }
+  }
 
-        final List<List<Contentlet>> listOfLists = new ArrayList<>();
-        final int batchSize = Config.getIntProperty("INDEX_COMMIT_LISTENER_BATCH_SIZE", 50);
-        for (final DotRunnable runner : listeners) {
-            if (runner instanceof ReindexRunnable) {
-                ReindexRunnable rrunner = (ReindexRunnable) runner;
-                if (rrunner.getAction().equals(ReindexRunnable.Action.REMOVING)) {
-                    rrunner.run();
-                    continue;
-                }
-                final List<Contentlet> cons = rrunner.getReindexIds();
-                for (Contentlet con : cons) {
-                    if (!reindexInodes.contains(con.getInode())) {
-                        reindexInodes.add(con.getInode());
-                        contentToIndex.add(con);
-                        if (contentToIndex.size() == batchSize) {
-                            listOfLists.add(contentToIndex);
-                            contentToIndex = new ArrayList<>();
-                        }
-                    }
-                }
-            } else {
-                runner.run();
-            }
+  private void internalRunner() {
+    final Set<String> reindexInodes = new HashSet<>();
+    List<Contentlet> contentToIndex = new ArrayList<>();
+
+
+    final List<List<Contentlet>> listOfLists = new ArrayList<>();
+    final int batchSize = Config.getIntProperty("INDEX_COMMIT_LISTENER_BATCH_SIZE", 50);
+    for (final DotRunnable runner : listeners) {
+      if (runner instanceof ReindexRunnable) {
+        ReindexRunnable rrunner = (ReindexRunnable) runner;
+        if (rrunner.getAction().equals(ReindexRunnable.Action.REMOVING)) {
+          rrunner.run();
+          continue;
         }
-        listOfLists.add(contentToIndex);
-        for (final List<Contentlet> batchList : listOfLists) {
-            new ReindexRunnable(batchList, ReindexRunnable.Action.ADDING, null, false) {
-            }.run();
+        final List<Contentlet> cons = rrunner.getReindexIds();
+        for (Contentlet con : cons) {
+          if (reindexInodes.add(con.getInode())) {
+            contentToIndex.add(con);
+            if (contentToIndex.size() == batchSize) {
+              listOfLists.add(contentToIndex);
+              contentToIndex = new ArrayList<>();
+            }
+          }
         }
+      } else {
+        runner.run();
+      }
     }
-}
\ No newline at end of file
+    listOfLists.add(contentToIndex);
+    for (final List<Contentlet> batchList : listOfLists) {
+      new ReindexRunnable(batchList, ReindexRunnable.Action.ADDING, null, false) {}.run();
+    }
+  }
+
+  private List<DotRunnable> getFlushers(final List<DotRunnable> allListeners) {
+    return allListeners.stream().filter(listener -> listener instanceof FlushCacheRunnable).collect(Collectors.toList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'listener' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)